### PR TITLE
Move pandas and numpy as integrations

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
I have attempted to refactor numpy and pandas as integrations rather than explicitly installing them/

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/numpy/__init__.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
```
Comment (schustmi) [inline src/zenml/integrations/numpy/__init__.py:1]:
```suggestion
#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
```

:P

File: src/zenml/integrations/numpy/__init__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Initialization of the NumPy integration."""
+
+from zenml.integrations.constants import NUMPY
+from zenml.integrations.integration import Integration
+
+
+class NumpyIntegration(Integration):
+    """Definition of NUMPY integration for ZenML."""
```
Comment (schustmi) [inline src/zenml/integrations/numpy/__init__.py:21]:
```suggestion
    """Definition of NumPy integration for ZenML."""
```

**Review by schustmi — CHANGES_REQUESTED** [review]
One other thing that is missing here is adding `numpy/pandas` to the `required_integrations` for all examples/blog posts that require them

**Review by fa9r — CHANGES_REQUESTED** [review]
Implementation so far looks good.

One thing that is still missing is to adjust the requirements of all other integrations that depend on numpy or pandas being installed. E.g., almost all data validator, feature store, and visualizer integrations do `import pandas as pd`, and some of the model deployers do `import numpy as np`, which might or might not work depending on whether the respective tool itself has numpy/pandas as a requirement. To be save, I would just add numpy/pandas as a requirement in the respective integration init.

**Comment by htahir1** [comment]
Closing because of internal discussions that we keep numpy and pandas as core dependencies for now

**Pull request closed without merging**